### PR TITLE
docs(python): Add storage_options hint for azure

### PR DESCRIPTION
**Pull request by francesco086**
Following issue https://github.com/pola-rs/polars/issues/18931.

I've tried to find a good place to put some hints in the documentation, but I can't find a good place in the polars documentation without ruining the nice minimalist style.

I am trying with a PR to add it in the only place I consider acceptable.

### CODE REVIEW

**Comment by francesco086** [comment]
One last comment from a user perspective.

Azure users are used to `adlfs` and it is extremely common (and best practice for security) to use the `anon=False` option to use an identity based authentication. What is nice about it, is that it will resolve automatically which identity to use.

Object Store does not offer the same functionality, which is really a pity, and therefore switching to polars has been quite confusing for me.
The solution I found is not as good as what `adlfs` offers, but it's the best I found.

In conclusion, I think many Azure users who want to use polars keeping in mind best security standards will encounter this issue, and I think some hints could spare them some troubles.

**Comment by ritchie46** [comment]
I think this is good, but could you do the same for gcp and aws?

**Comment by francesco086** [comment]
> I think this is good, but could you do the same for gcp and aws?

I agree that it would make a lot of sense and I would love to contribute, but I have always worked with Azure and am not familiar with gcp and aws. I have had a look at their respective ConfigKey and for example I could not see the CLI option.
So I think it would be better if someone who works with these two clouds could contribute.
